### PR TITLE
En_Wiz_Fire

### PR DESCRIPTION
**Pull request by Kelebek1**
Before opening this PR, ensure the following:
- `./format.sh` was run to apply standard formatting.
- `make` successfully builds a matching ROM.
- No new compiler warnings were introduced during the build process.
    - Can be verified locally by running `tools/warnings_count/check_new_warnings.sh`
- New variables & functions should follow standard naming conventions.
- Comments and variables have correct spelling.
---

### CODE REVIEW

File: src/overlays/actors/ovl_En_Wiz_Fire/z_en_wiz_fire.h [hunk 1]
```diff
@@ -7,10 +7,48 @@ struct EnWizFire;
 
 typedef void (*EnWizFireActionFunc)(struct EnWizFire*, GlobalContext*);
 
+typedef struct {
+    /* 0x00 */ u8 unk_00;
+    /* 0x01 */ u8 unk_01;
+    /* 0x02 */ char unk02[0x4];
```
Comment (jpburnett) [inline src/overlays/actors/ovl_En_Wiz_Fire/z_en_wiz_fire.h:13]:
```suggestion
    /* 0x02 */ UNK_TYPE1 unk02[0x4];
```

Comment (jpburnett) [reply src/overlays/actors/ovl_En_Wiz_Fire/z_en_wiz_fire.h:13]: Don't know if it makes a difference to have one over the other.

File: src/overlays/actors/ovl_En_Wiz_Fire/z_en_wiz_fire.h [hunk 2]
```diff
@@ -7,10 +7,48 @@ struct EnWizFire;
...
+    /* 0x06 */ s16 unk_06;
+    /* 0x08 */ s16 unk_08;
+    /* 0x0A */ char unk0A[0x2];
```
Comment (jpburnett) [inline src/overlays/actors/ovl_En_Wiz_Fire/z_en_wiz_fire.h:16]:
```suggestion
    /* 0x0A */ UNK_TYPE1 unk0A[0x2];
```

same as above

**Review by jpburnett — COMMENTED** [review]
Just some minor comments on char vs unk_type, good work overall!

**Review by engineer124 — COMMENTED** [review]
Nice, just some small comments from me

File: src/overlays/actors/ovl_En_Wiz_Fire/z_en_wiz_fire.c [hunk 3]
```diff
@@ -35,51 +51,838 @@ const ActorInit En_Wiz_Fire_InitVars = {
...
+        this->unk_168 = 100;
+    }
+    this->unk_160 = 0;
+    this->actionFunc = func_80A49A44;
+}
+
+void func_80A49A44(EnWizFire* this, GlobalContext* globalCtx) {
+    Vec3f sp54 = { 0.0f, 0.0f, 0.0f };
+
+    this->actor.world.rot.z += 0x1388;
```
Comment (AngheloAlf) [inline src/overlays/actors/ovl_En_Wiz_Fire/z_en_wiz_fire.c:175]: This is a round number
```suggestion
    this->actor.world.rot.z += 5000;
```

File: src/overlays/actors/ovl_En_Wiz_Fire/z_en_wiz_fire.c [hunk 4]
```diff
@@ -35,51 +51,838 @@ const ActorInit En_Wiz_Fire_InitVars = {
...
+        Matrix_StatePush();
+        Matrix_InsertTranslation(this->actor.world.pos.x, this->actor.floorHeight, this->actor.world.pos.z,
+                                 MTXMODE_NEW);
+        Matrix_NormalizeXYZ(&globalCtx->mf_187FC);
+
+        gSPSegment(POLY_XLU_DISP++, 0x08,
+                   Gfx_TwoTexScroll(globalCtx->state.gfxCtx, 0, globalCtx->state.frames & 0x7F,
+                                    (-globalCtx->state.frames * 6) & 0xFF, 0x20, 0x40, 1,
+                                    (globalCtx->state.frames * 2) & 0x7F, (-globalCtx->state.frames * 6) & 0xFF, 0x20,
+                                    0x40));
```
Comment (AngheloAlf) [inline src/overlays/actors/ovl_En_Wiz_Fire/z_en_wiz_fire.c:700]: Would this match?
```suggestion
                   Gfx_TwoTexScroll(globalCtx->state.gfxCtx, 0, globalCtx->state.frames % 128,
                                    (-globalCtx->state.frames * 6) % 256, 0x20, 0x40, 1,
                                    (globalCtx->state.frames * 2) % 128, (-globalCtx->state.frames * 6) %256, 0x20,
                                    0x40));
```

File: src/overlays/actors/ovl_En_Wiz_Fire/z_en_wiz_fire.c [hunk 5]
```diff
@@ -35,51 +51,838 @@ const ActorInit En_Wiz_Fire_InitVars = {
...
+        gSPMatrix(POLY_XLU_DISP++, Matrix_NewMtx(globalCtx->state.gfxCtx), G_MTX_NOPUSH | G_MTX_LOAD | G_MTX_MODELVIEW);
+        gSPDisplayList(POLY_XLU_DISP++, D_06003640);
+
+        Matrix_StatePop();
+        Matrix_InsertTranslation(this->actor.world.pos.x, this->actor.floorHeight, this->actor.world.pos.z, 0);
+        Matrix_NormalizeXYZ(&globalCtx->mf_187FC);
+
+        gSPSegment(POLY_XLU_DISP++, 0x08,
+                   Gfx_TwoTexScroll(globalCtx->state.gfxCtx, 0, (-globalCtx->state.frames * 3) & 0x7F, 0, 0x20, 0x20, 1,
+                                    0, (-globalCtx->state.frames * 10) & 0xFF, 0x20, 0x40));
```
Comment (AngheloAlf) [inline src/overlays/actors/ovl_En_Wiz_Fire/z_en_wiz_fire.c:716]: Same here
```suggestion
                   Gfx_TwoTexScroll(globalCtx->state.gfxCtx, 0, (-globalCtx->state.frames * 3) % 128, 0, 0x20, 0x20, 1,
                                    0, (-globalCtx->state.frames * 10) % 256, 0x20, 0x40));
```

File: src/overlays/actors/ovl_En_Wiz_Fire/z_en_wiz_fire.c [hunk 6]
```diff
@@ -35,51 +51,838 @@ const ActorInit En_Wiz_Fire_InitVars = {
...
+    OPEN_DISPS(globalCtx->state.gfxCtx);
+
+    Matrix_InsertTranslation(this->actor.world.pos.x, this->actor.floorHeight + 20.0f, this->actor.world.pos.z,
+                             MTXMODE_NEW);
+    Matrix_Scale(this->unk_14C + this->unk_154, this->unk_14C + this->unk_158, this->unk_14C + this->unk_15C,
+                 MTXMODE_APPLY);
+
+    gSPSegment(POLY_XLU_DISP++, 0x08,
+               Gfx_TwoTexScroll(globalCtx->state.gfxCtx, 0, 0, 0, 0x20, 0x20, 1, this->unk_170 & 0x7F,
+                                (-this->unk_170 * 10) & 0x7F, 0x20, 0x20));
```
Comment (AngheloAlf) [inline src/overlays/actors/ovl_En_Wiz_Fire/z_en_wiz_fire.c:789]:
```suggestion
               Gfx_TwoTexScroll(globalCtx->state.gfxCtx, 0, 0, 0, 0x20, 0x20, 1, this->unk_170 % 128,
                                (-this->unk_170 * 10) % 128, 0x20, 0x20));
```

**Review by AngheloAlf — APPROVED** [review]

File: src/overlays/actors/ovl_En_Wiz_Fire/z_en_wiz_fire.c [hunk 7]
```diff
@@ -35,51 +42,838 @@ const ActorInit En_Wiz_Fire_InitVars = {
...
+    s16 idx;
+
+    Actor_SetScale(&this->actor, this->unk_14C);
+    func_80A4BDDC(this, globalCtx);
+    this->unk_204 = this->unk_200 / 60.0f;
+
+    if (this->unk_162 == 0) {
+        EnWiz* wiz = (EnWiz*)this->actor.parent;
+
+        if ((wiz != NULL) && (wiz->actor.id == ACTOR_EN_WIZ) && (wiz->actor.update != NULL) && (wiz->unk_74A != 2)) {
```
Comment (hensldm) [inline src/overlays/actors/ovl_En_Wiz_Fire/z_en_wiz_fire.c:476]: Kind of minor, but I would prefer this variable just being type Actor, then cast it to `EnWiz` when checking `unk_74A`. As an example of something similar see `func_80BD9234` in `z_en_pamera`.

**Review by hensldm — APPROVED** [review]
Looks good. Have just one minor kind of nitpick, so going to approve.

**Review by Kenix3 — APPROVED** [review]

**Pull request merged**